### PR TITLE
chore: use msi in az copy command

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
use msi in az copy command.
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: e2e/windows/e2e-helper.sh [hunk 1]
```diff
@@ -68,19 +68,18 @@ upload_linux_file_to_storage_account() {
     E2E_MC_RESOURCE_GROUP_NAME="MC_${E2E_RESOURCE_GROUP_NAME}_${AZURE_E2E_CLUSTER_NAME}_$AZURE_BUILD_LOCATION"
     MC_VMSS_NAME=$(az vmss list -g $E2E_MC_RESOURCE_GROUP_NAME --query "[?contains(name, 'nodepool')]" -ojson | jq -r '.[0].name')
     VMSS_INSTANCE_ID="$(az vmss list-instances --name $MC_VMSS_NAME -g $E2E_MC_RESOURCE_GROUP_NAME | jq -r '.[0].instanceId')"
+    VMSS_RESOURCE_ID="$(az vmss show --name $MC_VMSS_NAME --resource-group $E2E_MC_RESOURCE_GROUP_NAME --instance-id $VMSS_INSTANCE_ID | jq '.id')"
+    az vmss identity assign --identities "${AZURE_MSI_RESOURCE_STRING}" -g $E2E_MC_RESOURCE_GROUP_NAME -n $MC_VMSS_NAME
+    az vmss update-instances -g $E2E_MC_RESOURCE_GROUP_NAME -n $MC_VMSS_NAME --instance-ids "*"
```
Comment (AbelHu) [inline e2e/windows/e2e-helper.sh:73]: updating instance should take a long time. Is it possible to create the vmss with setting the identity?

Comment (junjiezhang1997) [reply e2e/windows/e2e-helper.sh:73]: In my test, updating instance doesn't take too much time. But I think I could take a try to create the vmss with setting the identity.

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

File: e2e/windows/e2e-helper.sh [hunk 2]
```diff
@@ -68,19 +68,16 @@ upload_linux_file_to_storage_account() {
     E2E_MC_RESOURCE_GROUP_NAME="MC_${E2E_RESOURCE_GROUP_NAME}_${AZURE_E2E_CLUSTER_NAME}_$AZURE_BUILD_LOCATION"
     MC_VMSS_NAME=$(az vmss list -g $E2E_MC_RESOURCE_GROUP_NAME --query "[?contains(name, 'nodepool')]" -ojson | jq -r '.[0].name')
     VMSS_INSTANCE_ID="$(az vmss list-instances --name $MC_VMSS_NAME -g $E2E_MC_RESOURCE_GROUP_NAME | jq -r '.[0].instanceId')"
+    VMSS_RESOURCE_ID="$(az vmss show --name $MC_VMSS_NAME --resource-group $E2E_MC_RESOURCE_GROUP_NAME --instance-id $VMSS_INSTANCE_ID | jq '.id')"
```
Comment (AbelHu) [inline e2e/windows/e2e-helper.sh:71]: Do we still need this?

**Pull request merged**